### ISSUE TITLE
Publishing the copyrights and licenses of the third party libraries

### ISSUE THREAD

**Issue by mkauf** [envoyproxy/envoy#22874]
Some of the third party libraries that Envoy uses have license terms that demand that the copyright or the license itself is included in the software and/or documentation.

For example, [the nghttp2 library uses the MIT license](https://github.com/nghttp2/nghttp2/blob/master/COPYING). It contains this sentence: "Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to [...], subject to the following conditions: The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software."

How does Envoy comply with this legal requirement?

I think the copyright notice and the license of nghttp2 (and other third-party libraries) should appear somewhere in Envoy's documentation. There is already a [table that lists all third-party libraries](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/external_deps). A similar table that contains a list of the external dependencies, their licenses and (if necessary) also the complete license texts would solve this problem.

**Comment by mattklein123** [comment]
Yeah I'm sure we are not compliant to this degree. I will mark this help wanted in case someone would like to pick this up. Optimally this would all be automated in some way. (It would actually be nice of the license of each dependency was tracked and enforced to be filled in in our dependency metadata.)

cc @phlax

**Comment by phlax** [comment]
for github deps we can pick up the primary license at least - that should be pretty doable - altho as discussed recently it doesnt really handle the case of a repo with mixed licensing

in terms of including the actual copyright text - that is slightly more complex if it contains anything non-standard - altho i guess for most we can use the stock license text.

not sure exactly what our legal obligations are in terms of including the license text (IANAL etc) but i would agree including copyright information in our deps table in the docs seems to be a good way to do so

**Comment by phlax** [comment]
i guess the complexity is who the copyright holders are - not just what the license is - im not sure of any easy way to automate that but we could add a field in bazel/repo_locations.bzl to add it where it is known

**Comment by zhxie** [comment]
I may take a try to add a license field in `repository_locations.bzl` and render it on docs.

**Comment by mkauf** [comment]
Thank you for adding the licenses to the external dependencies table!

It would be great if transitive dependencies would also be listed - does somebody have this information?

For example, the [Common Expression Language (CEL) C++ library](https://github.com/google/cel-cpp) uses [ANTLR](https://github.com/antlr/antlr4). I have learned this because an envoy build failed a few weeks ago: The ANTLR project temporarily removed a git tag that the CEL library uses. See [here](https://github.com/google/cel-cpp/issues/161) and [here](https://github.com/antlr/antlr4/issues/3854). Without this build failure, I would not know that ANTLR is a transitive dependency of envoy.

**Comment by phlax** [comment]
ideally at least - this is exceptional - we do try to pin all of the deps explicitly in general

in the case of antlr it was listed in repository_locations until recently when we transitioned from using `rules_antlr` - not sure why its not there any more

cc @moderation @kyessenov

**Comment by kyessenov** [comment]
ANTLR is pulled transitively via this call https://github.com/google/cel-cpp/blob/master/bazel/deps.bzl#L70. I thought we had some verification steps that listed and compared all transitive dependencies? Maybe it's because ANTLR is a JAR file?
